src/wrappers/themis/rust/Cargo.toml

```diff
@@ -39,5 +36,4 @@ log = "0.4.6"
 env_logger = "0.6.0"
 
 [package.metadata.docs.rs]
-features = ["vendored"]
 dependencies = ["libssl-dev"]
```

Comment:
This is one of the original reasons why `libthemis-src` was added (see #349 and ilammy/rust-themis#9). The thing is, docs.rs does not have Themis installed so if the build requires Themis to be installed, then docs.rs will not be able to build API documentation.

I suspect that it's no longer necessary given that `libthemis-sys` now does not use Bindgen to generate bindings on-the-fly. However, I have checked and it seems that we still (kinda) need Themis to be installed because `build.rs` of `libthemis-sys` checks that Themis is installed, even when just `cargo doc` is run.

We can't remove `libthemis-src` without fixing up docs.rs generation. I have looked around and it seems that they [set `DOCS_RS` environment variable](https://github.com/rust-lang/docs.rs/pull/852) when generating documentation. We can use it in `libthemis-sys/build.rs`: look at that variable, if it is `1` then do not detect Themis installation and exit right away.

That way docs.rs will be able to run `cargo doc` without Themis being installed. In fact, we can remove this entire section because we don't need OpenSSL to be installed if we aren't going to build Themis.

It would also be nice to test this on CI somehow, but I don't see an immediate approach to that.

Reply:
Just checked, with
```rust
    if std::env::var("DOCS_RS") == Ok("1".to_string()) {
        return;
    }
```
in `libthemis-sys/build.rs -> main()`, `DOCS_RS=1 cargo doc` really works w/o libthemis installed, while running only `cargo doc` fails as expected. BTW, do we still need
```
[package.metadata.docs.rs]
dependencies = ["libssl-dev"]
```
?

Reply:
> `DOCS_RS=1 cargo doc` really works w/o libthemis installed

Amazing! I believe that should do the trick.

> do we still need `dependencies = ["libssl-dev"]`

No, I believe the entire `[package.metadata.docs.rs]` block can be removed. `libssl-dev` was required for Themis to compile, but if it's not going to be compiled then we don't need OpenSSL either.

Reply:
> It would also be nice to test this on CI somehow, but I don't see an immediate approach to that.

Tried passing different `PKG_CONFIG_*` env vars (described in `man pkg-config`) to `cargo build` to make the compilation fail while the libs are installed. No success. If so, I would add something like
`cargo clean --doc && DOCS_RS=1 PKG_CONFIG_SEARCH_ONLY_IN=/nonexistingdir cargo doc --workspace --no-deps`
to `run_tests.sh`

Reply:
Another issue with CI is that libthemis-sys will try the system installation if pkg-config fails to find Themis Code. CI builds and installs Themis Core into the system before running RustThemis tests. We will need to somehow uninstall it for the docs.rs tests and then install it back for other tests. Maybe a better idea will present itself later. For now, I think, we can leave it like that. It's enough that you have tested it manually.